src/Ocelot/Cache/CacheKeyGenerator.cs

```diff
@@ -1,23 +1,39 @@
-﻿using System.Text;
-using System.Threading.Tasks;
-
+﻿using Ocelot.Configuration;
 using Ocelot.Request.Middleware;
+using System;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
 
 namespace Ocelot.Cache
 {
     public class CacheKeyGenerator : ICacheKeyGenerator
     {
-        public string GenerateRequestCacheKey(DownstreamRequest downstreamRequest)
+        public string GenerateRequestCacheKey(DownstreamRequest downstreamRequest, DownstreamRoute downstreamRoute)
         {
-            var downStreamUrlKeyBuilder = new StringBuilder($"{downstreamRequest.Method}-{downstreamRequest.OriginalString}");
+            var builder = new StringBuilder($"{downstreamRequest.Method}-{downstreamRequest.OriginalString}");
+
+            var cacheOptionsHeader = downstreamRoute?.CacheOptions?.Header;
+
+            if (!string.IsNullOrEmpty(cacheOptionsHeader))
+            {
+                var header = downstreamRequest.Headers
+                    .FirstOrDefault(r => r.Key.Equals(cacheOptionsHeader, StringComparison.OrdinalIgnoreCase))
+                    .Value?.FirstOrDefault();
+
+                if (!string.IsNullOrEmpty(header))
+                {
+                    builder.Append(header);
+                }
+            }
+
```

Comment:
Because of this significant changes:
- Do we have unit tests for the `GenerateRequestCacheKey` method?
- Could you add unit tests to cover this new logic/lines please?

Reply:
resolve